src/Validator/Constraints/PaymentMethodCheckoutValidator.php

```diff
@@ -62,4 +63,13 @@ public function validate($value, Constraint $constraint): void
         }
         $this->context->buildViolation($constraint->message)->setTranslationDomain('messages')->addViolation();
     }
+
+    private function isMollieGatewayFactory(GatewayConfigInterface $gateway): bool
```

Comment:
I saw somewhere below same code but with no private method separation. How about creating trait for this?